meshtal_analysis.py

```diff
@@ -5,28 +5,93 @@
 """
 import matplotlib as mpl
 import matplotlib.pyplot as plt
-import matplotlib.colors as colors
+from matplotlib import colors
 import numpy as np
 import argparse
+import math
 import logging as ntlogger
 import pandas as pd
-import neut_utilities as ut
 mpl.use('Agg')
 
 
 class meshtally:
-    idnum = None
-    ptype = None
-    x_bounds = None
-    y_bounds = None
-    z_bounds = None
-    e_bounds = None
-    t_bounds = None
-    data = None
-    x_mids = None
-    y_mids = None
-    z_mids = None
-    ctype = None
+    """Mesh tally object data"""
+    def __init__(self):
+        self.idnum = None
+        self.ptype = None
+        self.x_bounds = []
+        self.y_bounds = []
+        self.z_bounds = []
+        self.e_bounds = []
+        self.t_bounds = []
+        self.data = []
+        self.x_mids = []
+        self.y_mids = []
+        self.z_mids = []
+        self.ctype = None
+
+    def __str__(self):
+        info = ("\n Number of voxels: " + str(self.number_voxels()) +
+                " \n Number of bins in: \n x-dimension- " +
+                str(len(self.x_mids)) + " \n y-dimension- " +
+                str(len(self.y_mids)) + "\n z-dimension- " +
+                str(len(self.z_mids)) + "\n voxel volume: " +
+                str(self.voxel_volume()) + " cm squared")
+        return info
+
+    def number_voxels(self):
+        """
+        Member function of meshtally. gets the number of midpoints to
+        find number of voxels.
+        Return: int
+        """
+        num_x = len(self.x_mids)
+        num_y = len(self.y_mids)
+        num_z = len(self.z_mids)
+        return (num_x * num_y * num_z)
+
+    def voxel_volume(self):
+        """ Member Function of meshtally. check uniform and finds volume
+        from distance to adjacent vertex.
+        Raises:
+            ValueError: voxels not uniform
+        Returns:
+            float: volume of voxel
+        """
+        if (check_uniform(self.x_bounds) and check_uniform(self.y_bounds) and
+            check_uniform(self.z_bounds)):
+
+            x = abs(float(self.x_bounds[1]) - float(self.x_bounds[0]))
+            y = abs(float(self.y_bounds[1]) - float(self.y_bounds[0]))
+            z = abs(float(self.z_bounds[1]) - float(self.z_bounds[0]))
+
+            return x * y * z
+        else:
+            raise ValueError("voxels not uniform")

```

Comment:
as discussed we don't want this to error as most meshes are not uniform, probably should either give average volume or just state non-uniform